tests/Routes/Wiki/DeletedWikiMetricsControllerTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Tests\Routes\Wiki;
+use App\Http\Controllers\WikiController;
+use App\WikiManager;
+use Carbon\Carbon;
+use Carbon\CarbonImmutable;
+use Tests\Routes\Traits\OptionsRequestAllowed;
+use Tests\TestCase;
+use App\WikiSiteStats;
+use App\WikiSetting;
+use App\Wiki;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+class DeletedWikiMetricsControllerTest extends TestCase
+{
+    protected string $route = 'deletedWikiMetrics';
+
+    use OptionsRequestAllowed;
+    use DatabaseTransactions;
+
+    public function setUp(): void {
+        parent::setUp();
+        Wiki::query()->delete();
+        WikiSiteStats::query()->delete();
+        WikiSetting::query()->delete();
+    }
+
+    public function tearDown(): void {
+        Wiki::query()->delete();
+        WikiSiteStats::query()->delete();
+        WikiSetting::query()->delete();
+        parent::tearDown();
+    }
+
+    public function testDownloadCsv()
+    {
+        $this->createAndDeleteTestWiki('one.wikibase.cloud', 0, '',1, 2);
+        $this->createAndDeleteTestWiki( 'two.wikibase.cloud', 0, 'Some Reason', 0, 3 );
+        $response = $this->get($this->route);
+        $response->assertStatus(200)
+            ->assertDownload(CarbonImmutable::now()->toIso8601String().'-deleted_wiki_metric.csv');
+    }
+
+    private function createAndDeleteTestWiki($name, $user_id, $wikiDeletionReason, $createdWeeksAgo = 1, $wiki_users = 1) {
+        $current_date = CarbonImmutable::now();
+
+        $wiki = Wiki::factory()->create([
+            'domain' => $name, 'sitename' => 'bsite'
+        ]);
+        WikiManager::factory()->create( [
+            'wiki_id' => $wiki->id, 'user_id' => $user_id,
+        ] );
+        WikiSiteStats::factory()->create([
+            'wiki_id' => $wiki->id, 'pages' => 77, 'users' => $wiki_users
+        ]);
+        $wiki->created_at = $current_date->subWeeks($createdWeeksAgo);
+
+        $wiki->save();
+        WikiController::delete($this->get($this->route));
```

Comment:
You probably meant to call this api using the test harness in a similar to what you did on line 40 in this patch. You can see a similar thing at tests/Routes/QsBatch/QsControllerTest.php#73 .

Reply:
I think it could be better not to depend on the WikiController here. If the Controller method breaks then this test will break too and we will be confused.

Why don't we just run the same logic that would happen if we were to delete the wiki using the controller. Delete the Wiki and then have two cases: one where we add a reason and one where we don't?